### DISCUSSION TITLE
Termux access to /data/data

### DISCUSSION THREAD

**Discussion by SpeedyCraftah** [termux/termux-app#3856] [Q&A] [answered]
I was wondering how Termux is able to access all directories under `/data/data` when using root, when other apps (root apps) can only see "com.google.android.gms" and the app directory itself, so how does Termux circumvent this restriction? I'm looking to be able to access data in `/data/data` of other apps under root in my personal app, I also checked and see some other root file explorers have this restriction as well, so surely Termux can't be special and must be using some work around to bypass this sandboxing? I would assume this is related to SELinux.

Thanks!

**Comment by Biswa96** [comment]
> I was wondering how Termux is able to access all directories under `/data/data` when using root

Are you running termux as root? Wouldn't it be obvious that any program can access those directories with root permission?

**Reply by SpeedyCraftah** [reply]
I am, I am also running my app as root, even confirmed with a whoami. I was surprised as   well that I can't see all of the data directories when running my app as root, but Termux can do it no problem, so I'm not sure what's going on here.

This also applies to some root file Explorer apps I saw, so this must be some new permission change in the Android system for newer versions.

**Comment by agnostic-apollo** [comment, marked as answer]
That's because of app data isolation that's engaged if an app uses `targetSdkVersion` `>= 30` (Android `11`) in which `/data/data/*` of other apps do not exist in the mount namespace of the app, and only its own and specific whitelisted packages like `gms`. Termux app uses `28` currently so it's not isolated. You can disable it for all apps by running `su -c "setprop persist.zygote.app_data_isolation false"` and rebooting, at the cost of privacy. Or in other apps, run `su` shell with the `-mm`/`--mount-master` flag.

https://cs.android.com/android/platform/superproject/+/android-14.0.0_r1:frameworks/base/services/core/java/com/android/server/am/ProcessList.java;l=2305

https://cs.android.com/android/platform/superproject/+/android-14.0.0_r1:frameworks/base/core/jni/com_android_internal_os_Zygote.cpp;l=1186

**Reply by SpeedyCraftah** [reply]
Thank you very much!

**Reply by agnostic-apollo** [reply]
Welcome.

**Reply by marcussacana** [reply]
doubt, seems su -c "setprop persist.zygote.app_data_isolation false" did not worked for me but su --mm worked, is there a way to launch an external app on the device with this mount-master?

**Reply by agnostic-apollo** [reply]
Did you restart? And running other apps from root or adb shell should still work with `am`.

**Reply by marcussacana** [reply]
of course, but since I have strong root hidding tools installed it may some of then ignoring my setting, I will test am, thx

**Reply by lamekaayo123** [reply]
Howw
